### PR TITLE
Cli: More precise detection of invaild arguments

### PR DESCRIPTION
**Pull request by zenhack**
I think @naved001 just hit this while testing #683, and I think I've bumped into it on a branch where I'm working on #232. The solution is simple enough.

See the commit message for more overview.

### CODE REVIEW

File: haas/cli.py [hunk 1]
```diff
@@ -135,25 +135,29 @@ def cmd(f):
...
+    showee = [f.__name__] + ['<%s>' % name for name in args]
+    args = ' '.join(['<%s>' % name for name in args])
+    if varargs:
+        showee += ['<%s...>' % varargs]
+    usage_dict[f.__name__] = ' '.join(showee)
+
     @wraps(f)
     def wrapped(*args, **kwargs):
         try:
+            if len(args) < num_args or not varargs and len(args) > num_args:
```
Comment (knikolla) [inline haas/cli.py]: Can you please provide an explanation/comment for this, especially the ``not varargs`` part?

**Comment by zenhack** [comment]
Quoting Kristi Nikolla (2017-03-01 15:04:27)

>    Can you please provide an explanation/comment for this, especially the
>    not varargs part?

Done.

**Comment by zenhack** [comment]
I assume the test failures were because of the problem solved by #730. I just merged that to master and pulled it back in here.

**Comment by knikolla** [comment]
Squash-merged to remove the merge commit from history.

**Comment by knikolla** [comment]
@zenhack Happy 2000th commit of HIL.

**Pull request merged**